sdk/migrationdiscoverysap/Azure.ResourceManager.MigrationDiscoverySap/tests/Tests/MigrationSapDiscoveryTests.cs

```diff
@@ -0,0 +1,113 @@
+        Assert.AreEqual(ProvisioningState.Succeeded, sapDiscoverySiteResource.Data.ProvisioningState);
+        string sapDiscoverySiteId = sapDiscoverySiteResource.Id.ToString();
+
+        // Upload Excel to SAP DiscoverySite
+        string excelPathToImport = @"TestData\ExcelSDKTesting.xlsx";
+        await SapDiscoveryTestsHelpers.PostImportEntities(sapDiscoverySiteResource, excelPathToImport, Client);
+
+        // Get List SAP Instances
+        List<SAPInstanceData> listSapInstances = await SapDiscoveryTestsHelpers
+            .GetSapInstancesListAsync(sapDiscoverySiteResource);
+
+        Assert.IsNotNull(listSapInstances);
+
+        const string expectedSapInstancesListPath = @"TestData\ExpectedGetSapInstanceList.json";
+
+        List<SAPInstanceData> expectedListSapInstances = SapDiscoveryTestsHelpers.GetExpectedSapInstancesListFromJson(
+            sapDiscoverySiteId,
+            expectedSapInstancesListPath);
+
+        listSapInstances = listSapInstances.OrderBy(instance => instance.Id.ToString().ToLower()).ToList();
+        expectedListSapInstances = expectedListSapInstances.OrderBy(instance => instance.Id.ToString().ToLower()).ToList();
+
+        Assert.IsTrue(SapDiscoveryTestsHelpers.AreSapInstancesListEqual(listSapInstances, expectedListSapInstances));
+
+        // Get SapInstance
+        ResourceIdentifier sapInstanceId = listSapInstances.First().Id;
+        Assert.IsNotNull(sapInstanceId);
+        SAPInstanceResource sapInstance = await sapDiscoverySiteResource.GetSAPInstanceAsync(sapInstanceId.Name);
+
+        // Get Server Instances List
+
+        List<ServerInstanceData> serverInstancesList = await SapDiscoveryTestsHelpers.GetServerInstancesList(sapInstance);
+        Assert.IsNotNull(serverInstancesList);
+
+        const string expectedServerInstanceListPath = @"TestData\ExpectedGetServerInstancesList.json";
+
+        List<ServerInstanceData> expectedServerInstancesList = SapDiscoveryTestsHelpers
+            .GetExpectedServerInstancesListFromJson(sapDiscoverySiteId, expectedServerInstanceListPath);
+
+        serverInstancesList = serverInstancesList.OrderBy(server => server.Name.ToLower()).ToList();
+        expectedServerInstancesList = expectedServerInstancesList.OrderBy(server => server.Name.ToLower()).ToList();
+
+        Assert.IsTrue(SapDiscoveryTestsHelpers.AreServerInstancesListEqual(serverInstancesList, expectedServerInstancesList));
+
+        // Get ServerInstance
+        ResourceIdentifier serverInstanceId = serverInstancesList.First().Id;
+        ServerInstanceResource serverInstance = await sapInstance.GetServerInstanceAsync(serverInstanceId.Name);
+
+        //Patch SAP DiscoverySite
+        var discoverySitePatch = new SAPDiscoverySitePatch();
+        discoverySitePatch.Tags.Add("Key1", "TestPatchValue");
+        sapDiscoverySiteResource = await sapDiscoverySiteResource.UpdateAsync(discoverySitePatch);
+        Assert.AreEqual(discoverySiteName, sapDiscoverySiteResource.Data.Name);
+
```

Comment:
should we check Tags value after patching, why check names?

Reply:
Done